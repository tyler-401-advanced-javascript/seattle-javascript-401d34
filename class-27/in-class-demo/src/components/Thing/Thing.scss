.Thing {
  background-color: yellow;
  font: {
    family: Arial;
    size: 3em;
  }
  button {
    display: block;
    margin-top: 2em;
  }
}
