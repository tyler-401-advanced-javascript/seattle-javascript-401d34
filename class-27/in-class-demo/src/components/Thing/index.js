import React from 'react'
import './Thing.scss'

class Thing extends React.Component {
  constructor () {
    super()
    this.state = {
      stuff: true
    }
  }

  handleChange = e => {
    this.setState({
      stuff: !this.state.stuff
    })
  }

  render () {
    return (
      <section className="Thing">
        <p>My wonderful component</p>
        <span>Stuff: {this.state.stuff.toString()}</span>
        <button id="myButton" onClick={this.handleChange}>Clicky</button>
      </section>
    )
  }
}

export default Thing
