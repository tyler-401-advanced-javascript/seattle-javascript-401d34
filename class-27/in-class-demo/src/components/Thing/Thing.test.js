import React from 'react'
import Thing from './'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'

describe('<Thing />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Thing />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('changes state on click', () => {
    const component = mount(<Thing />)
    const button = component.find('#myButton')
    button.simulate('click')
    expect(component).toHaveState({ stuff: false })
    expect(component.find('span')).toIncludeText('false')
  })
})
