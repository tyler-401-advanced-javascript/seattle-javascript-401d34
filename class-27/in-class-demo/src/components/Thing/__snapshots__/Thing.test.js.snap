// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Thing /> renders correctly 1`] = `
<section
  className="Thing"
>
  <p>
    My wonderful component
  </p>
  <span>
    Stuff: 
    true
  </span>
  <button
    id="myButton"
    onClick={[Function]}
  >
    Clicky
  </button>
</section>
`;
